app/models/stacks_image.rb

```diff
@@ -1,10 +1,31 @@
 ##
 # Images in the image stacks
-class StacksImage < StacksFile
+class StacksImage
+  include BackedByFile
   include DjatokaAdapter
 
   attr_accessor :canonical_url, :size, :region, :rotation, :quality, :format
 
+  # @return [RestrictedImage] the restricted version of this image
+  def restricted
+    RestrictedImage.new(region: region,
+                        size: size,
+                        rotation: rotation,
+                        quality: quality,
+                        format: format,
+                        id: id,
+                        file_name: file_name)
+  end
+
+  # TODO: remove tight coupling to djatoka
```

Comment:
What are you thinking about this? Maybe getting the raw IIIF JSON hash out of djatoka and then directly modifying it?

Reply:
Yes, exactly.